resolvers/webpack/index.js

```diff
@@ -79,75 +82,137 @@ exports.resolve = function (source, file, settings) {
   log('Using config: ', webpackConfig)
 
   // externals
-  if (findExternal(source, webpackConfig.externals, path.dirname(file))) return { found: true, path: null }
-
-  var otherPlugins = []
-
-  // support webpack.ResolverPlugin
-  if (webpackConfig.plugins) {
-    webpackConfig.plugins.forEach(function (plugin) {
-      if (plugin.constructor && plugin.constructor.name === 'ResolverPlugin' && Array.isArray(plugin.plugins)) {
-        otherPlugins.push.apply(otherPlugins, plugin.plugins);
-      }
-    });
+  if (findExternal(source, webpackConfig.externals, path.dirname(file))) {
+    return { found: true, path: null }
   }
 
   // otherwise, resolve "normally"
-  var resolver = createResolver(webpackConfig.resolve || {}, otherPlugins)
+  var resolveSync = createResolveSync(configPath, webpackConfig)
   try {
-    return { found: true, path: resolver.resolveSync(path.dirname(file), source) }
+    return { found: true, path: resolveSync(path.dirname(file), source) }
   } catch (err) {
     log('Error during module resolution:', err)
     return { found: false }
   }
 }
 
-var Resolver = require('enhanced-resolve/lib/Resolver')
+function createResolveSync(configPath, webpackConfig) {
+  var webpackRequire
+
+  try {
+    var webpackFilename = resolve.sync('webpack', { basedir: path.dirname(configPath) })
+    var webpackResolveOpts = { basedir: path.dirname(webpackFilename) }
+
+    webpackRequire = function (id) {
+      return require(resolve.sync(id, webpackResolveOpts))
+    }
+  } catch (e) {
+    // Something has gone wrong (or we're in a test). Use our own bundled
+    // enhanced-resolve.
+    webpackRequire = require
+  }
+
+  var enhancedResolvePackage = webpackRequire('enhanced-resolve/package.json')
+  var enhancedResolveVersion = enhancedResolvePackage.version
+  log('enhanced-resolve version:', enhancedResolveVersion)
+
+  var resolveConfig = webpackConfig.resolve || {}
+
+  if (semver.major(enhancedResolveVersion) === 2) {
+    return createWebpack2ResolveSync(webpackRequire, resolveConfig)
+  }
+
+  return createWebpack1ResolveSync(webpackRequire, resolveConfig, webpackConfig.plugins)
+}
+
+function createWebpack2ResolveSync(webpackRequire, resolveConfig) {
+  var EnhancedResolve = webpackRequire('enhanced-resolve')
 
-var SyncNodeJsInputFileSystem = require('enhanced-resolve/lib/SyncNodeJsInputFileSystem')
-var syncFS = new SyncNodeJsInputFileSystem()
+  return EnhancedResolve.create.sync(assign({}, resolveConfig, webpack2DefaultResolveConfig))
```

Comment:
I think the order of `resolveConfig` and `webpack2DefaultResolveConfig` should be reversed, so that the custom resolveConfig overrides the values on webpack2DefaultResolveConfig. Great job!